SetupCurrencyConversions asks for history data ahead of Now time

<!--- This template provides sections for bugs and features. Please delete any irrelevant sections before submitting -->

#### Expected Behavior
Cannot ask for history ahead of now time
#### Actual Behavior
<!--- Required. Describe the actual behavior for your case. -->
Fails to start Lean, because start date is ahead of now date, failing with the following error

`20190408 01:54:25.790 ERROR:: Engine.Run(): During the algorithm initialization, the following exception has occurred: Oanda.RestV20.Client.ApiException: Error calling GetInstrumentCandles: {"errorMessage":"Invalid value specified for 'from'. Time is in the future"}
  at Oanda.RestV20.Api.DefaultApi.GetInstrumentCandlesWithHttpInfo (System.String authorization, System.String instrument, System.String acceptDatetimeFormat, System.String price, System.String granularity, System.Nullable`1[T] count, System.String from, System.String to, System.Nullable`1[T] smooth, System.Nullable`1[T] includeFirst, System.Nullable`1[T] dailyAlignment, System.String alignmentTimezone, System.String weeklyAlignment) [0x00362] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Brokerages/Oanda/RestV20/Api/DefaultApi.cs:3816 
  at Oanda.RestV20.Api.DefaultApi.GetInstrumentCandles (System.String authorization, System.String instrument, System.String acceptDatetimeFormat, System.String price, System.String granularity, System.Nullable`1[T] count, System.String from, System.String to, System.Nullable`1[T] smooth, System.Nullable`1[T] includeFirst, System.Nullable`1[T] dailyAlignment, System.String alignmentTimezone, System.String weeklyAlignment) [0x00001] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Brokerages/Oanda/RestV20/Api/DefaultApi.cs:3735 
  at QuantConnect.Brokerages.Oanda.OandaRestApiV20+<DownloadQuoteBars>d__25.MoveNext () [0x00084] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Brokerages/Oanda/OandaRestApiV20.cs:566 
  at System.Collections.Generic.List`1[T].AddEnumerable (System.Collections.Generic.IEnumerable`1[T] enumerable) [0x00059] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:1108 
  at System.Collections.Generic.List`1[T]..ctor (System.Collections.Generic.IEnumerable`1[T] collection) [0x00062] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:87 
  at System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x0000e] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/ToCollection.cs:30 
  at QuantConnect.Brokerages.Oanda.OandaBrokerage+<GetHistory>d__14.MoveNext () [0x000f6] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Brokerages/Oanda/OandaBrokerage.cs:228 
  at QuantConnect.Lean.Engine.DataFeeds.Enumerators.QuoteBarFillForwardEnumerator.MoveNext () [0x00001] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Engine/DataFeeds/Enumerators/QuoteBarFillForwardEnumerator.cs:70 
  at QuantConnect.Lean.Engine.DataFeeds.Enumerators.FillForwardEnumerator.MoveNext () [0x00088] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Engine/DataFeeds/Enumerators/FillForwardEnumerator.cs:137 
  at QuantConnect.Lean.Engine.DataFeeds.SubscriptionData+<Enumerator>d__8.MoveNext () [0x00066] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Engine/DataFeeds/SubscriptionData.cs:80 
  at QuantConnect.Lean.Engine.DataFeeds.Subscription.MoveNext () [0x00010] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Engine/DataFeeds/Subscription.cs:201 
  at QuantConnect.Lean.Engine.HistoricalData.BrokerageHistoryProvider.GetHistory (System.Collections.Generic.IEnumerable`1[T] requests, NodaTime.DateTimeZone sliceTimeZone) [0x0002f] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Engine/HistoricalData/BrokerageHistoryProvider.cs:71 
  at QuantConnect.Lean.Engine.Setup.BaseSetupHandler.SetupCurrencyConversions (QuantConnect.Interfaces.IAlgorithm algorithm, QuantConnect.Lean.Engine.DataFeeds.UniverseSelection universeSelection) [0x001b7] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Engine/Setup/BaseSetupHandler.cs:92 
  at QuantConnect.Lean.Engine.Setup.BrokerageSetupHandler.Setup (QuantConnect.Lean.Engine.Setup.SetupHandlerParameters parameters) [0x005ac] in /Users/erikbengtson/Documents/Projects/erikbengtson/Lean/Engine/Setup/BrokerageSetupHandler.cs:354 Error calling GetInstrumentCandles: {"errorMessage":"Invalid value specified for 'from'. Time is in the future"}
`
#### Potential Solution
<!--- Optional. Describe any potential solutions and/or thoughts as to what may be causing the difference between expected and actual behavior. -->

#### Reproducing the Problem
<!--- Required for Bugs. Describe how to reproduce the problem. This can be via a failing unit test or a simplified algorithm that reliably demonstrates this issue.  -->
Start lean on Sunday night (before monday 12am utc). If you start lean on monday 12am, then problem does not occur

#### System Information
<!--- Required for Bugs. Include any system specific information, such as OS. -->
macOS 

#### Checklist
<!--- Confirm that you've provided all the required information. -->
<!--- Required fields --->
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
<!--- Required for Bugs, feature request can delete the line below. -->
- [x] I have provided detailed steps to reproduce the issue

<!--- Template inspired by https://github.com/stevemao/github-issue-templates -->

--- a/Algorithm/QCAlgorithm.cs
+++ b/Algorithm/QCAlgorithm.cs
@@ -959,8 +959,19 @@ namespace QuantConnect.Algorithm
             // the time rules need to know the default time zone as well
             TimeRules.SetDefaultTimeZone(timeZone);
 
-            // reset the current time according to the time zone
-            SetDateTime(_startDate.ConvertToUtc(TimeZone));
+            // In BackTest mode we reset the Algorithm time to reflect the new timezone
+            // startDate is set by the user so we expect it to be for their timezone already
+            // so there is no need to update it.
+            if (!LiveMode)
+            {
+                SetDateTime(_startDate.ConvertToUtc(TimeZone));
+            }
+            // In live mode we need to adjust startDate to reflect the new timezone
+            // startDate is set by Lean to the default timezone (New York), so we must update it here
+            else
+            {
+                _startDate = DateTime.UtcNow.ConvertFromUtc(TimeZone).Date;
+            }
         }
 
         /// <summary>
@@ -1362,7 +1373,8 @@ namespace QuantConnect.Algorithm
                 Securities.SetLiveMode(live);
                 if (live)
                 {
-                    _startDate = DateTime.Today;
+                    // startDate is set relative to the algorithm's timezone.
+                    _startDate = DateTime.UtcNow.ConvertFromUtc(TimeZone).Date;
                     _endDate = QuantConnect.Time.EndOfTime;
                 }
             }

